libsignal-service/src/sender.rs

```diff
@@ -518,11 +519,14 @@ where
             match send {
                 Ok(SendMessageResponse { needs_sync }) => {
                     tracing::debug!("message sent!");
-                    return Ok(SentMessage {
-                        recipient,
-                        unidentified: unidentified_access.is_some(),
-                        needs_sync,
-                    });
+                    return Ok((
+                        SentMessage {
+                            recipient,
+                            unidentified: unidentified_access.is_some(),
+                            needs_sync,
+                        },
+                        identity_key,
+                    ));
```

Comment:
Would it make sense to add `identity_key` to `SentMessage` instead?

Reply:
```patch
diff --git a/libsignal-service/src/sender.rs b/libsignal-service/src/sender.rs
index 0e23fc05c..adb02c7c7 100644
--- a/libsignal-service/src/sender.rs
+++ b/libsignal-service/src/sender.rs
@@ -49,14 +49,14 @@ pub struct SendMessageResponse {
     pub needs_sync: bool,
 }
 
-pub type SendMessageResult =
-    Result<(SentMessage, IdentityKey), MessageSenderError>;
+pub type SendMessageResult = Result<SentMessage, MessageSenderError>;
 
 #[derive(Debug, Clone)]
 pub struct SentMessage {
     pub recipient: ServiceAddress,
     pub unidentified: bool,
     pub needs_sync: bool,
+    pub identity_key: IdentityKey,
 }
 
 /// Attachment specification to be used for uploading.
@@ -347,7 +347,11 @@ where
             // if we sent a data message and we have linked devices, we need to send a sync message
             (
                 ContentBody::DataMessage(message),
-                Ok((SentMessage { needs_sync, .. }, _identity_key)),
+                Ok(SentMessage {
+                    needs_sync,
+                    identity_key: _,
+                    ..
+                }),
             ) if *needs_sync || self.is_multi_device().await => {
                 tracing::debug!("sending multi-device sync message");
                 let sync_message = self
@@ -420,7 +424,11 @@ where
                 .await;
 
             match result {
-                Ok((SentMessage { needs_sync, .. }, _)) if needs_sync => {
+                Ok(SentMessage {
+                    needs_sync,
+                    identity_key: _,
+                    ..
+                }) if needs_sync => {
                     needs_sync_in_results = true;
                 },
                 _ => (),
@@ -518,14 +526,12 @@ where
             match send {
                 Ok(SendMessageResponse { needs_sync }) => {
                     tracing::debug!("message sent!");
-                    return Ok((
-                        SentMessage {
-                            recipient,
-                            unidentified: unidentified_access.is_some(),
-                            needs_sync,
-                        },
+                    return Ok(SentMessage {
+                        recipient,
+                        unidentified: unidentified_access.is_some(),
+                        needs_sync,
                         identity_key,
-                    ));
+                    });
                 },
                 Err(ServiceError::Unauthorized)
                     if unidentified_access.is_some() =>
@@ -908,14 +914,14 @@ where
                     let SentMessage {
                         recipient,
                         unidentified,
+                        identity_key,
                         ..
-                    } = sent.0;
-                    let identity_key = sent.1;
+                    } = sent;
                     UnidentifiedDeliveryStatus {
                         destination_service_id: Some(
                             recipient.uuid.to_string(),
                         ),
-                        unidentified: Some(unidentified),
+                        unidentified: Some(*unidentified),
                         destination_identity_key: Some(
                             identity_key.serialize().into(),
                         ),

```

Reply:
With the patch above (on top of the PNI branch), Whisperfish compiles and runs fine (on top of the PNI branch), after a quick test at least.

`UnidentifiedDeliveryStatus` doesn't show up in the PNI branch, but the changes need to be checked in - I'd say that is another PR later.

Reply:
That's a very good catch, didn't even occur to me. See a1caf30f5.